code/controllers/subsystems/dbcore.dm

```diff
@@ -0,0 +1,554 @@
+	if(SSdbcore.Connect())
+		for(var/datum/db_query/query in queries_current)
+			run_query(query)
+
+		var/datum/db_query/query_round_shutdown = SSdbcore.NewQuery(
+			"UPDATE [format_table_name("round")] SET shutdown_datetime = Now(), end_state = :end_state WHERE id = :round_id",
+			list("end_state" = SSticker.end_game_state, "round_id" = game_id)
+		)
+		query_round_shutdown.Execute()
+		qdel(query_round_shutdown)
+	if(IsConnected())
+		Disconnect()
+
+// Implementation OOS
+// /datum/controller/subsystem/dbcore/can_vv_get(var_name)
+// 	if(var_name == NAMEOF(src, connection))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, all_queries))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_active))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_new))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_standby))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_active))
+// 		return FALSE
+
+// 	return ..()
+
+// /datum/controller/subsystem/dbcore/vv_edit_var(var_name, var_value)
+// 	if(var_name == NAMEOF(src, connection))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, all_queries))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_active))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_new))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_standby))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_active))
+// 		return FALSE
+// 	return ..()
+
+/datum/controller/subsystem/dbcore/proc/Connect()
+	if(IsConnected())
+		return TRUE
+
+	if(failed_connection_timeout <= world.time) //it's been more than 5 seconds since we failed to connect, reset the counter
+		failed_connections = 0
+
+	if(failed_connections > 5) //If it failed to establish a connection more than 5 times in a row, don't bother attempting to connect for 5 seconds.
+		failed_connection_timeout = world.time + 5 SECONDS
+		return FALSE
+
+	if(!sqlenabled)
+		return FALSE
+
+	var/user = sqllogin
+	var/pass = sqlpass
+	var/db = sqldb
+	var/address = sqladdress
+	var/port = text2num(sqlport)
+	//var/timeout = max(CONFIG_GET(number/async_query_timeout), CONFIG_GET(number/blocking_query_timeout)) // Commented out until it matters
+	var/timeout = max(10, 5)
+	//var/thread_limit = CONFIG_GET(number/bsql_thread_limit) // See above
+	var/thread_limit = 50
+
+	//max_concurrent_queries = CONFIG_GET(number/max_concurrent_queries) // See above
+	max_concurrent_queries = 25 // TODO: Make the above matter
+
+	var/result = json_decode(rustg_sql_connect_pool(json_encode(list(
+		"host" = address,
+		"port" = port,
+		"user" = user,
+		"pass" = pass,
+		"db_name" = db,
+		"read_timeout" = timeout,
+		"write_timeout" = timeout,
+		"max_threads" = thread_limit,
+	))))
+	. = (result["status"] == "ok")
+	if (.)
+		connection = result["handle"]
+	else
+		connection = null
+		last_error = result["data"]
+		log_sql("Connect() failed | [last_error]")
+		++failed_connections
```

Comment:
Rather complex, would this be more concise?

```suggestion
	if (result["status"] == "ok")
		connection = result["handle"]
		return TRUE
	else
		connection = null
		last_error = result["data"]
		log_sql("Connect() failed | [last_error]")
		++failed_connections
```

Reply:
Changes made